### ISSUE TITLE
Similarity Tool

### ISSUE THREAD

**Issue by sbembenek18** [Electrostatics/apbs#140]
While I am able to successfully use the similarity tool over a range of small molecules, when going beyond a grid size of 65 (for the input DX files), for example 97, 161, I consistently get a segmentation fault.

**Comment by sbembenek18** [comment]
I wanted to follow up on this and see if this issue has been resolved. thanks.

**Comment by sobolevnrm** [comment]
Sorry; I have not done anything about this yet.

**Comment by sbembenek18** [comment]
No worries. Thanks.

**Comment by sbembenek18** [comment]
Following up on this. It doesn't seem to be simply a grid size issue as I just got the segmentation fault error for grid size of 65 and had to drop it to 33 to get the similarity calculation to run correctly. Look forward to your thoughts. Let me know if sending the DX file would be useful (can't attach here).

**Comment by sobolevnrm** [comment]
Yes, providing the DX file and a description of how you're invoking the code would be very helpful.  You should be able to ZIP the DX file and attach it here.  Thanks.

**Comment by sbembenek18** [comment]
I am running the command as (per the attached file) --> 

similarity --format dx --scalar1 ref_mol_pymol-generated_65.dx --scalar2 ref_mol_pymol-generated_65.dx

(Just a quick check against itself.) When the grid is 65 it gives a segmentation fault (see attached output.txt), but runs fine when the grid is remade to be 33. I had a similar experience with other molecules where lowering the grid size fixed the issue. In general, I have not been able to go past a grid size of 65. 

The DX file is made through Pymol using the APBS plugin (I have attached the resulting APBS input file).

Thanks. for looking into it. Let me know if I can provide more info. 

[output.txt](https://github.com/Electrostatics/apbs/files/8801085/output.txt)
[pymol-generated.in.tar.gz](https://github.com/Electrostatics/apbs/files/8801086/pymol-generated.in.tar.gz)
[ref_mol_pymol-generated_65.dx.tar.gz](https://github.com/Electrostatics/apbs/files/8801087/ref_mol_pymol-generated_65.dx.tar.gz)

**Comment by sbembenek18** [comment]
Quick follow up in this: we're you able to re-produce the error?

**Comment by sobolevnrm** [comment]
I have not had time to try; sorry.  I am also not sure when I will have time to try.  I am keeping this issue open in the hope that I will get time in the future but the similarity tool has never been a key feature of the code and, as such, this has lower priority than other issues.

**Comment by sbembenek18** [comment]
OK. Thanks.
